pkg/services/ngalert/api/hooks.go

```diff
@@ -11,39 +13,48 @@ import (
 type RequestHandlerFunc func(*contextmodel.ReqContext) response.Response
 
 type Hooks struct {
-	logger log.Logger
-	hooks  map[string]RequestHandlerFunc
+	logger     log.Logger
+	router     *mux.Router
+	routeHooks map[*mux.Route]RequestHandlerFunc
 }
 
 // NewHooks creates an empty set of request handler hooks. Hooks can be used
 // to replace handlers for specific paths.
 func NewHooks(logger log.Logger) *Hooks {
 	return &Hooks{
-		logger: logger,
-		hooks:  make(map[string]RequestHandlerFunc),
+		logger:     logger,
+		router:     mux.NewRouter(),
+		routeHooks: make(map[*mux.Route]RequestHandlerFunc),
```

Comment:
We unfortunately have to keep this map, because mux only knows how to attach `http.Handler`s to routes. Grafana uses a completely custom HTTP layer, so the handlers look different - Mux does not understand them.